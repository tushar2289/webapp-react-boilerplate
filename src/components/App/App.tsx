import React from 'react';
import {
  createStyles,
  makeStyles,
  ThemeProvider,
  createMuiTheme,
} from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import useMediaQuery from '@material-ui/core/useMediaQuery';
import AppHeader from '../AppHeader';
import Home from '../Home';
import About from '../About';
import { Provider, rootStore } from '../../models/Root';

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      flexGrow: 1,
    },
  })
);

const App: React.FC = () => {
  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');

  const theme = React.useMemo(
    () =>
      createMuiTheme({
        palette: {
          type: prefersDarkMode ? 'dark' : 'light',
        },
      }),
    [prefersDarkMode]
  );
  const classes = useStyles();
  return (
    <Provider value={rootStore}>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <div className={classes.root}>
          <BrowserRouter>
            <AppHeader />
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/about" component={About} />
            </Switch>
          </BrowserRouter>
        </div>
      </ThemeProvider>
    </Provider>
  );
};

export default App;
