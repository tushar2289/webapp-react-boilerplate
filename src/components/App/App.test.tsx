import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import App from './App';

afterEach(cleanup);

test('renders learn react link', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('open about page', () => {
  const { getByText } = render(<App />);
  fireEvent.click(getByText('About'));
  expect(getByText('Source code')).toBeInTheDocument();
});
