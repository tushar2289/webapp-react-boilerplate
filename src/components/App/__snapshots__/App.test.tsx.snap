// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-3 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
        >
          Material UI
        </h6>
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                class="makeStyles-linkButton-4"
                href="/"
              >
                Home
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                class="makeStyles-linkButton-4"
                href="/about"
              >
                About
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <h1
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Create react app boilerplate
        </h1>
        <div
          class="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root-7"
          >
            <div
              class="MuiBox-root MuiBox-root-8"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                 MobX powered counter
              </h6>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Decrement
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <span
              class="MuiBox-root MuiBox-root-9"
            >
              <span
                class="MuiTypography-root MuiTypography-h6"
              >
                0
              </span>
            </span>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Increment
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
